credsweeper/filters/cred_card_number_check.py

```diff
@@ -0,0 +1,39 @@
+from credsweeper.credentials import LineData
+from credsweeper.filters import Filter
+
+
+class CreditCardNumberCheck(Filter):
+    """Check that value is a credit card number."""
+
+    def run(self, line_data: LineData) -> bool:
+        """Run filter checks on received credential candidate data 'line_data'.
+
+        Args:
+            line_data: credential candidate data
+
+        Return:
+            False, if the sequence is not card number. True if it is
+
+        """
+        if line_data.value is None \
+                or 16 != len(line_data.value) \
+                or ("0" == line_data.value[0] and "0" == line_data.value[1]):
+            return True
+        try:
+            s = 0
+            # https://en.wikipedia.org/wiki/Luhn_algorithm
+            for n in range(0, 16):
+                x = int(line_data.value[n])
+                if not 1 & n:
+                    x *= 2
+                    if x > 9:
+                        x -= 9
+                s += x
+
+            if 0 == s % 10:
+                return False
```

Comment:
Could you explain it more?
I can't understand even I read the wiki page.
The code block is different with pusedo code from wiki page.

Reply:
You might be confused. 
I used the algorithm, where the size is 16 (regex in rule is fixed). e.g. https://www.groundlabs.com/blog/anatomy-of-a-credit-card/
Then, move "from left to right". The wiki pseudocode has opposite direction and the original Luhn algorithm is designed for various size of number.
The approach may be used for 15 symbols cards with zero-left-padding.
E.g. AmericanExpress 378282246310005 - > 0378282246310005

Reply:
Okay. I understood little more not fully..
As I see the pattern detect only 16 symbols but the code include 15 symbols or started with `00` cases.
If some card number which is 15 symbols typed like as you explained above, okay i agree.
But in `00` case.. why is it needed? for 14 symbols cases?

Reply:
many dumps for 64bit offset have the signature and many cases
```
0000000000000000 n wm4.cstdlib.42.38c5efcdb0f7c179bd11e611a0b94359
```
the check filters the case much quickly

Reply:
00 cannot start the card number but there are may core dumps with memory addresses in this format, aren't they?

Reply:
why don't you leave the check for just one 0 in the beginning? It seems enough and it may being checked in the regex.

Reply:
@kmnls 
i assume, threre might be an american express card number with padding. Then left character will be ``0``.
Sequense ``40xx...`` makes a regex too complicated. e.g. [1-9]{2}[0-9]{14} - does not work.